content/java/concepts/array-list/terms/indexOf/indexOf.md

```diff
@@ -0,0 +1,92 @@
+---
+Title: '.indexOf()'
+Description: 'If ArrayList contains given element returns the index of the first occurrence of it , else -1'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`.indexOf()`** method returns the index of the first occurrence of the specified element in [`ArrayList`](https://www.codecademy.com/resources/docs/java/array-list), or -1 if it does not contain the element
+
+## Syntax
+
+```pseudo
+arrayListInstance.indexOf(element);
+```
+
+Signature of the method is `indexOf(Object element)` where `element` is the element whose lowest index is to be returned.
+
+it does not throw any exception if the element is not present in the [`ArrayList`](https://www.codecademy.com/resources/docs/java/array-list) but returns -1.
+
+but `.equals()` method may throw a NullPointerException if the element is null.
+
+## Example
+
+```java
+// Import the ArrayList class from the java.util package
+import java.util.ArrayList;
+
+public class Main {
+  public static void main(String[] args) {
+    // Create ArrayList of type String named animals
+    ArrayList<String> animals = new ArrayList<String>();
+    // Add elements to ArrayList refer .add() docs for more info
+    animals.add("Lion");
+    animals.add("Tiger");
+    animals.add("Cat");
+    animals.add("Dog");
+    animals.add("Tiger");
+    animals.add("Lion");
+    animals.add("Tiger");
+    // Print the index of the first occurrence of the element "Tiger" in this case 1
+    System.out.println(animals.indexOf("Tiger\n"));
+    System.out.println(animals.indexOf("Elephant"));
+
+  }
+}
+```
+
+The code above will print the index of the first occurrence of the element `"Tiger"` in the [`ArrayList`](https://www.codecademy.com/resources/docs/java/array-list) which is `1`.
+
+## Output
+
+```shell
+1
+-1
+```
+
+The `.indexOf()` method does not work with primitive data types.
```

Comment:
I'm not sure I agree with this point. Isn't it more so that `.indexOf()` is not compatible with _regular arrays_ because it is an `ArrayList` method?

Reply:
```
// Online Java Compiler
// Use this editor to write, compile and run your Java code online
import java.util.ArrayList;
class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello, World!");
        ArrayList<int> list = new ArrayList<>();  
        list.add(1);
        list.add(2);
        list.add(3);

int index = list.indexOf(2);  
System.out.println(index);

        
    }
}
```
##Output wud be err
```
javac /tmp/CmHIzfwn03/HelloWorld.java
/tmp/CmHIzfwn03/HelloWorld.java:7: error: unexpected type
ArrayList<int> list = new ArrayList<>();  
                  ^
  required: reference
  found:    int
1 error

```

Reply:
```
// Online Java Compiler
// Use this editor to write, compile and run your Java code online
import java.util.ArrayList;
class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello, World!");
        ArrayList<Integer> list = new ArrayList<>();  
        list.add(1);
        list.add(2);
        list.add(3);

int index = list.indexOf(2);  
System.out.println(index);

        
    }
}
```
##output normal as we r using integer
```
java -cp /tmp/CmHIzfwn03 HelloWorld
Hello, World!1
```

Reply:
Integer is an object whereas int is primitive data type
below pic is from official java docs
![image](https://user-images.githubusercontent.com/92514129/210931561-61e1b5b7-484f-422a-abc0-976729fe52a5.png)
as you can see what .indexOf supports is object not a primitive data type as mentioned in docs...